sdk/core/core/src/az_span.c

```diff
@@ -618,3 +618,74 @@ _az_span_scan_until(az_span self, _az_predicate predicate, int32_t* out_index)
   }
   return AZ_ERROR_ITEM_NOT_FOUND;
 }
+
+AZ_NODISCARD az_span _az_span_trim_white_space(az_span source)
+{
+  // Trim from end after trim from start
+  return _az_span_trim_white_space_from_end(_az_span_trim_white_space_from_start(source));
+}
+
+AZ_NODISCARD AZ_INLINE bool _az_is_white_space(uint8_t c)
+{
+  switch (c)
+  {
+    case ' ':
+    case '\t':
+    case '\n':
+    case '\r':
+      return true;
+  }
+  return false;
+}
+
+typedef enum
+{
+  LEFT = 0,
+  RIGHT = 1,
+} az_span_trim_side;
+
+// Return a trim az_span. Depending on arg side, function will trim left of right
+AZ_NODISCARD static az_span _az_span_trim_side(az_span source, az_span_trim_side side)
```

Comment:
Up to you, but I preferred the separate methods you had before (which imo captured the intent more clearly).

As a code reader, I have to maintain more state to know if the slices are correct now since the methods are merged.

Also disassembly size grows from 61 to 87 (on clang). Did you notice code size improvements with this change?

It looks like it is more code and code size now than what was there before:
https://godbolt.org/z/KLYFt_

Clang (default)
231 vs 240 instructions
18760 vs 43806 bytes

Clang (-Os)
117 vs 130 instructions
21536 vs 96938 bytes

Reply:
I like to avoid the code duplication.
Please add more details about the disasembly (how did you do it.  I wouldn't mind about a tiny increment, but those numbers looks weird...  9 more instructions ended up on more than the double of bytes...

Reply:
> how did you do it

The godbolt link I shared shows the generated assembly instructions and you can count the number of instructions (I just copied into diff checker to get the number of instructions difference):
https://www.diffchecker.com/uX8IIJFz

I believe the bytes are "size on disk", but even if we ignore that, I think the simplicity of having two functions and the smaller disassembly generated is worth the trade off, to me.

If you prefer, write a C small library or app that uses the trim APIs and check the size difference when you compile it. Also, check the before/after of the library size for azure core with your change vs what I am suggestion.

![image](https://user-images.githubusercontent.com/6527137/81017031-6699c680-8e16-11ea-8895-4f3e58eb118f.png)

Reply:
this is an internal implementation, I prefer to keep it as one

Reply:
Alright, even if it is potentially less efficient in terms of code size/runtime perf?

Reply:
We can always come back in the future and decide if this needs to be changed.
For now, I prefer how one single function looks. 
Specially removing that decremental loop from the trim_end().

I see both solutions quite simple enough to better investigate on other functions on the SDK to improve performance. We might want to make a full analysis on performance (as function testing) to detect and fix top offenders instead of using time to investigate each function alone (unit testing)